package duke.playlist;
option java_package = "fr.mikrosimage.player.playlist";
option java_outer_classname = "PlaylistProtocol";

message Playlist {
	repeated Shot shot 					= 1;
	optional Display display 			= 2;
	optional AudioSource audioSource	= 3;
	optional uint32 startFrame			= 4;
	optional bool loop					= 5 [ default = true ];
	optional float framerate 			= 6 [ default = 25 ];
}

message Shot {
	required string media 				= 1;
	optional uint32 mediaStart 			= 2;
	optional uint32 mediaEnd			= 3;
	optional uint32 trackStart 			= 4;
	optional uint32 trackEnd			= 5;
	optional Display display 			= 6;
	optional string track 				= 7 [ default = "default" ];
	optional bool reverse 				= 8 [ default = false ];
}

message Display {
	enum ColorSpace {
		LIN								= 0;
		SRGB							= 1;
		LOG								= 2;
	}
	optional ColorSpace colorspace		= 1;
//	optional string lut 				= 2;
}

message AudioSource {
	required string name 				= 1;
}